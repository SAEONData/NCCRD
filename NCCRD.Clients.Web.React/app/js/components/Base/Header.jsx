'use strict'

import React from 'react'
import { Col, Row, Container } from 'mdbreact';

//Images
import environmental_affairs_logo from '../../../images/environmental_affairs_logo.png'
import sa_flag from '../../../images/sa_flag.jpg'
import nrf_seaon from '../../../images/nrf_saeon.png'

class Header extends React.Component {

  constructor(props) {
    super(props);
  }
  render() {

    return (
      <>
        <Container>
          <Row className="align-items-center" style={{ marginBottom: "15px" }}>
            <Col md="3" className="d-none d-md-block">
              <img src={environmental_affairs_logo} style={{ width: "100%", marginTop: "2%", marginLeft: "-10px" }} align="left" valign="middle" />
            </Col>
            <Col md="6">
              <h1 style={{ textAlign: "center", marginTop: "4%", letterSpacing: "2px", color: "#2e7d32" }}>
                <b>NCCRD</b>
              </h1>
              <h5 style={{ textAlign: "center", letterSpacing: "2px", marginTop: "-8px", color: "grey" }}>
                <b>National Climate Change Response Database</b>
              </h5>
            </Col>
            <Col md="3" className="d-none d-md-block">
              <img src={nrf_seaon} style={{ width: "90%", marginTop: "4%", marginRight: "0px" }} align="right" />
            </Col>
          </Row>
        </Container>
      </>
    )
  }
}

export default Header