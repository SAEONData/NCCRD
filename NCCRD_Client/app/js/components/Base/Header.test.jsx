import React from "react";
import { Col, Row } from 'mdbreact';
import { mount, shallow } from "enzyme";
import Header from "./Header.jsx";

describe("Header", () => {
  let mountedHeader;
  const header = () => {
    if (!mountedHeader) {
      mountedHeader = mount(
        <Header />
      );
    }
    return mountedHeader;
  }

  beforeEach(() => {
    mountedHeader = undefined
  });

  // All tests will go here
  it('should render correctly with no props', () => {
    //const component = shallow(<Header />)
    expect(header()).toMatchSnapshot()
  })

  it("always renders two images", () => {
    expect(header().find("img").length).toBe(2);
  })

  it("contains text 'NCCRD'", () => {
    expect(header().find("b").first().text()).toBe("NCCRD");
  })

  it("contains text 'National Climate Change Response Database'", () => {
    expect(header().find("b").last().text()).toBe("National Climate Change Response Database");
  })

  it("contains 1 Row", () => {
    expect(header().find(Row).length).toBe(1)
  })

  it("contains 5 Columns", () => {
    expect(header().find(Col).length).toBe(5)
  })
});