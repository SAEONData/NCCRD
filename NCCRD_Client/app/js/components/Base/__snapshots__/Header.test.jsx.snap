// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly with no props 1`] = `
<Header>
  <div
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <t
      className="align-items-center"
      style={
        Object {
          "height": 100,
        }
      }
      tag="div"
    >
      <div
        className="row align-items-center"
        style={
          Object {
            "height": 100,
          }
        }
      >
        <t
          className="d-none d-md-block"
          lg={null}
          md="2"
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col-md-2 d-none d-md-block"
          >
            <img
              align="left"
              onClick={[Function]}
              src={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "cursor": "pointer",
                  "height": 80,
                  "marginLeft": 5,
                }
              }
            />
          </div>
        </t>
        <t
          lg={null}
          md="1"
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col-md-1"
          />
        </t>
        <t
          lg={null}
          md="6"
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col-md-6"
          >
            <div>
              <h1
                style={
                  Object {
                    "color": "#2e7d32",
                    "letterSpacing": "2px",
                    "textAlign": "center",
                  }
                }
              >
                <b>
                  NCCRD
                </b>
                <sub
                  style={
                    Object {
                      "backgroundColor": "#78e26c",
                      "borderRadius": "5px",
                      "fontSize": "18px",
                      "padding": "2px",
                    }
                  }
                >
                  <i>
                    BETA
                  </i>
                </sub>
              </h1>
              <p />
              <h5
                style={
                  Object {
                    "color": "grey",
                    "letterSpacing": "2px",
                    "marginTop": "-8px",
                    "textAlign": "center",
                  }
                }
              >
                <b>
                  National Climate Change Response Database
                </b>
              </h5>
            </div>
          </div>
        </t>
        <t
          lg={null}
          md="1"
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col-md-1"
          />
        </t>
        <t
          className="d-none d-md-block"
          lg={null}
          md="2"
          sm={null}
          tag="div"
          xl={null}
          xs={null}
        >
          <div
            className="col-md-2 d-none d-md-block"
          >
            <img
              align="right"
              src={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 80,
                  "marginRight": 5,
                }
              }
            />
          </div>
        </t>
      </div>
    </t>
  </div>
</Header>
`;
